testapi.pm

```diff
@@ -1030,7 +1030,9 @@ C<$serialdev>.
 sub assert_script_sudo {    # no:style:signatures
     my ($cmd, $wait) = @_;
     my $str = hashed_string("ASS$cmd");
-    script_sudo("$cmd; echo $str-\$?- > /dev/$serialdev", 0);
+    my $redirect_to_serial_console = if is_serial_terminal() ? '' : "> /dev/$serialdev";
+    my $marker = sprintf "echo $str-\$?- %s", $redirect_to_serial_console;
```

Comment:
It should suffice to just do

```
my $marker = "echo $str-\$?-$redirect_to_serial_console";
```

Reply:
I would still add a space, though

Reply:
hm, the space should probably be added to the beginning of `$redirect_to_serial_console` and not at the end of `$marker`, though.

Reply:
i am not sure what's the problem with the sprintf? or it is not request (maybe just a personal preference?)

Reply:
Well, I assume there is unnecessary overhead when calling sprintf. sprintf might be required if the requirement is higher but here the simple variable expansion within strings work. This is Perl, not C :)

Reply:
*ready for review*